todo/src/actions/actions.js

```diff
@@ -0,0 +1,35 @@
+import uuid from 'uuid';
+
+export const ADD_TODO = 'ADD_TODO';
+export const TOGGLE_COMPLETED = 'TOGGLE_COMPLETED';
+export const DELETE_TODO = 'DELETE_TODO';
+export const DELETE_COMPLETED = 'DELETE_COMPLETED';
+
+export const addTodo = todo => {
+  return {
+    type: ADD_TODO,
+    payload: {
+      id: uuid.v4(),
+      value: todo,
+      completed: false
+    }
+  };
+};
+
+export const toggleCompleted = id => {
+  return {
+    type: TOGGLE_COMPLETED,
+    payload: id
```

Comment:
both for this and the deleteTodo actioncreator, you want to do exactly this: send the _minimum_ amount of info with the payload as possible to minimize errors within your reducer function, so just the id instead of the whole todo and then extracting the id from it later